src/node_file.cc

```diff
@@ -3118,6 +3118,130 @@ static void GetFormatOfExtensionlessFile(
   return args.GetReturnValue().Set(EXTENSIONLESS_FORMAT_JAVASCRIPT);
 }
 
+static void CpSyncCheckPaths(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK_EQ(args.Length(), 4);  // src, dest, dereference, recursive
+
+  BufferValue src(isolate, args[0]);
+  CHECK_NOT_NULL(*src);
+  ToNamespacedPath(env, &src);
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemRead, src.ToStringView());
+  auto src_path = std::filesystem::path(src.ToStringView());
```

Comment:
`BufferValue` converts the string to UTF8, on Windows this requires automatic conversion for char8_t which may not be supported by the version of MSVC we use, I think that may be the source of bugs like https://github.com/nodejs/node/issues/54476

Reply:
Actually no, this is simply broken on Windows, because this just gets returned in char. It needs to be returned as std::u8string instead.